kubernetes-azure-go/main.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"encoding/base64"
+
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/containerservice"
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/network"
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/resources"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		// Get some configuration values or set default values
+		cfg := config.New(ctx, "")
+		azureLocation := config.Require(ctx, "azure-native:location")
+		sshPubKey := cfg.Require("sshPubKey")
+		mgmtGroup := cfg.Require("mgmtGroupId")
+		prefixForDns, err := cfg.Try("prefixForDns")
+		if err != nil {
+			prefixForDns = "pulumi"
+		}
+		kubernetesVersion, err := cfg.Try("kubernetesVersion")
+		if err != nil {
+			kubernetesVersion = "1.24.3"
+		}
+		numWorkerNodes, err := cfg.TryInt("numWorkerNodes")
+		if err != nil {
+			numWorkerNodes = 3
+		}
+		nodeVmSize, err := cfg.Try("nodeVmSize")
+		if err != nil {
+			nodeVmSize = "Standard_DS2_v2"
+		}
+
+		// Create an Azure Resource Group
+		resourceGroup, err := resources.NewResourceGroup(ctx, "resourceGroup", &resources.ResourceGroupArgs{
+			Location:          pulumi.String(azureLocation),
+			ResourceGroupName: pulumi.String("aks-rg"),
```

Comment:
The `azureLocation` variable _is_ pulled from the provider configuration (see line 17).

As for the names, we can certainly remove those lines and let them be auto-generated. I left them there in the event the user wanted to specify a name, but perhaps letting them get auto-generated would be a better approach.

Reply:
> The azureLocation variable is pulled from the provider configuration (see line 17).

Oh yes -- what I mean is that I _think_ if you _removed_ both line 17 and the references to `Location` throughout, things would "just work", since all Azure Native resources should (by default) use whatever location is configured _on_ the provider.

> As for the names, we can certainly remove those lines and let them be auto-generated.

Ok cool, yeah I'd definitely recommend that then.

Reply:
Fixed via 270555d